src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.cs

```diff
@@ -75,19 +76,32 @@ public sealed partial class AnalyzerConfig
 
         internal ImmutableArray<Section> NamedSections { get; }
 
+        /// <summary>
+        /// Gets the concatenation of the <see cref="GlobalSection" /> and <see cref="NamedSections" />
+        /// </summary>
+        internal IEnumerable<Section> AllSections { get; }
+
         /// <summary>
         /// Gets whether this editorconfig is a topmost editorconfig.
         /// </summary>
         internal bool IsRoot => GlobalSection.Properties.TryGetValue("root", out string val) && val == "true";
 
-        private AnalyzerConfig(
+        /// <summary>
+        /// Gets whether this editorconfig is a merged global editorconfig
+        /// </summary>
+        internal bool IsMergedGlobal { get; }
```

Comment:
Same comment. I don't think we should use the AnalyzerConfig type to represent a bunch of merged global configs. For one, the path to file is now meaningless if this is true.

Let's use AnalyzerConfig to represent exactly what the user wrote, namely a single editorconfig file.

Reply:
I'll make another type and pass it directly to the set.

---
In reply to: [421833001](https://github.com/dotnet/roslyn/pull/43889#discussion_r421833001) [](ancestors = 421833001)